### ISSUE TITLE
PPO squared-error loss

### ISSUE THREAD

**Issue by tommyvsfu1** [JasonYao81000/ADL2019#1]
想請問下PPO value loss的計算方法，因為PPO paper上好像沒定義很清楚squared-error loss的計算( 有點不太能理解？)
![](https://i.imgur.com/AsrzJYl.png)

我發現你在 ADL, MLDS 兩邊的PPO squared-error loss 計算方式不大一樣，然後我看了網路上很多寫法(tensorlayer等等）也都不大相同，想請問下這方便你是怎麼理解的？謝謝

ADL hw3 (agent_pg.py)
loss = -torch.min(surr1, surr2) + 1.0 * **nn.MSELoss()(state_values, rewards)**- 0.01 * entropy

MLDS hw4 (agent_pg.py)
**loss_vf = tf.squared_difference(self.rewards + self.gamma * self.v_preds_next, v_preds)**

**Comment by yao-jason** [comment]
The pytorch version loss function is referenced from [here](https://github.com/nikhilbarhate99/PPO-PyTorch/blob/master/PPO_continuous.py#L123).
The algorithm we used on both ADL and MLDS is PPO2, which is different from the original PPO.
You can find the difference between them in the following figure, and learn more details in the online videos ([part 1](https://youtu.be/z95ZYgPgXOY), [part 2](https://youtu.be/OAKAZhFmYoI)) lectured by Hung-Yi Lee.
![image](https://user-images.githubusercontent.com/31233752/61792459-61ebbf00-ae4f-11e9-96b3-2e6cf15aac3d.png)
